File: 1877-minimize-maximum-pair-sum-in-array/1877-minimize-maximum-pair-sum-in-array.cs
public class Solution {
    public int MinPairSum(int[] a) {
        int MinSum = -1;
        Array.Sort(a);
        
        int l = 0 , r = a.Length - 1;
        while (l <= r){
            MinSum = Math.Max (MinSum , a[l ++] + a[r --]);
        }
        return MinSum;
    }
}

File: 1877-minimize-maximum-pair-sum-in-array/1877-minimize-maximum-pair-sum-in-array.cpp
class Solution {
public:
    int minPairSum(vector<int>& a) {
        int MinSum = INT_MIN;
        sort (a.begin() , a.end());
        int l = 0 , r = a.size() - 1;
        while (l <= r){
            MinSum = max (a[l++] + a[r--] , MinSum);
        }
        
        return MinSum;
    }
};